sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClient.java

```diff
@@ -0,0 +1,589 @@
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return registry artifact properties.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<RegistryArtifactProperties>> getRegistryArtifactPropertiesWithResponse(String tagOrDigest) {
+        return withContext(context -> this.getRegistryArtifactPropertiesWithResponse(tagOrDigest, context));
+    }
+
+    Mono<Response<RegistryArtifactProperties>> getRegistryArtifactPropertiesWithResponse(String tagOrDigest, Context context) {
+        try {
+            Mono<String> getTagMono = tagOrDigest.contains(":")
+                ? Mono.just(tagOrDigest)
+                : this.getTagProperties(tagOrDigest).map(a -> a.getDigest());
+
+            return getTagMono
+                .flatMap(tag -> this.serviceClient.getRegistryArtifactPropertiesWithResponseAsync(repositoryName, tag))
+                .map(res -> Utils.mapResponse(res, Utils::mapArtifactProperties))
+                .onErrorMap(Utils::mapException);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * <p>Get registry artifact properties.</p>
+     *
+     * <p>This method can take in both a digest as well as a tag.<br>
+     * In case a tag is provided it calls the service to get the digest associated with it.</p>
+     * @param tagOrDigest tag or digest associated with the blob.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return registry artifact properties.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<RegistryArtifactProperties> getRegistryArtifactProperties(String tagOrDigest) {
+        return this.getRegistryArtifactPropertiesWithResponse(tagOrDigest).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * List registry artifacts based of a repository.
+     *
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return manifest attributes.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<RegistryArtifactProperties> listRegistryArtifacts() {
+        return listRegistryArtifacts(null);
+    }
+
+    /**
+     * List manifests of a repository.
+     *
+     * @param options the options associated with the list registy operation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return manifest attributes.
```

Comment:
Let's use the same terminology in all docs. All other methods refer to this as `registry artifact properties`. We should use the same here.